### PR TITLE
Local git hooks

### PR DESCRIPTION
**Pull request by Ressetkk**
**Description**
Added directory and scripts with useful git hook that ease the development.

**Related issue(s)**

#3805

### CODE REVIEW

**Comment by Ressetkk** [comment]
Disabled SC2086 checking because git client cannot stat quoted value from env variable.

Workaround for Mac IDEs https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000180659-Pre-commit-hook-bug-on-NPM-when-using-VCS-tools
https://apple.stackexchange.com/questions/51677/how-to-set-path-for-finder-launched-applications

It's a known issue in all JetBrains programs.

File: .githooks/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Git hools
+
+## Overview
+
+This directory provides useful git hooks that can be used in standard git development flow.
```
Comment (klaudiagrz) [inline .githooks/README.md:5]:
```suggestion
This directory provides Git hooks that you can use in a standard Git development flow.
```

File: .githooks/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Installation
+
+To use those hooks in your have to configure the default git hooks path for your repository.
+Run the command below in the root of your `test-infra` git repository:
+```shell
+git config core.hooksPath .githooks
+```
+After that your local git repository will use git hooks from `.githooks` path.
```
Comment (klaudiagrz) [inline .githooks/README.md:14]:
```suggestion
Now your local Git repository uses Git hooks from the `.githooks` path.
```

File: .githooks/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Troubleshooting
+
+### Mac: VCS integration in GoLand can't use applications installed by Homebrew
+
+This is an issue that is known for all JetBrains IDEs which run on a Mac.
+Upon startup GoLand will initialize the `PATH` based on the macOS' `launchd` path.
```
Comment (klaudiagrz) [inline .githooks/README.md:21]:
```suggestion
Upon startup, GoLand will initialize the `PATH` based on the macOS' `launchd` path.
```

File: .githooks/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+The [solution](https://apple.stackexchange.com/questions/51677/how-to-set-path-for-finder-launched-applications) is to add `/usr/local/bin` path to the system `PATH` variable by using the command listed below.
+**This change will be applied to ALL uses of a computer!**
+```shell
+sudo launchctl config user path /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
+```
+
+After running this command reboot your machine to apply changes.
+
+## Hooks list
+|Name|Type|Description|
```
[the remark this answers is not in the document]
Comment (Ressetkk) [reply .githooks/README.md:31]: I assumed the will will only grow from this time so in order to not disrupt the length between overview and the rest of the instruction I placed it at the bottom as a separate section.

File: .githooks/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+### Mac: VCS integration in GoLand can't use applications installed by Homebrew
+
+This is an issue that is known for all JetBrains IDEs which run on a Mac.
+Upon startup, GoLand will initialize the `PATH` based on the macOS' `launchd` path.
+The [solution](https://apple.stackexchange.com/questions/51677/how-to-set-path-for-finder-launched-applications) is to add `/usr/local/bin` path to the system `PATH` variable by using the command listed below.
+```shell
+sudo launchctl config user path /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
+```
+**>CAUTION:** This change will be applied to ALL uses of a computer!
```
Comment (klaudiagrz) [inline .githooks/README.md:26]:
```suggestion
>**CAUTION:** This change will be applied to ALL uses of a computer!
```

**Review by klaudiagrz — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by adamwalach — APPROVED** [review]

**Pull request merged**